BUILDING

```diff
@@ -1 +1,13 @@
-To build Ion, use the build.sh or build.py scripts in the ion/ directory.
+1. Install dependencies
+  - gyp
+  - clang
+  - libc++-dev
+  - libc++abi-dev
+
```

Comment:
Any version that supports C++11 should be okay.

In your case, it looks like some kind of misconfiguration on your end. You are using clang, but it's pulling in GCC 5.2.1 C++ headers.